src/NuGet.Core/NuGet.Protocol/Resources/PackageMetadataResourceV3.cs

```diff
@@ -149,6 +149,11 @@ private async Task<T> DeserializeStreamDataAsync<T>(Stream stream, CancellationT
         {
             token.ThrowIfCancellationRequested();
 
+            if (stream == null)
+            {
+                return default(T);
```

Comment:
Test please? :)

Reply:
Added 2 unit test which correspond with below 2 cases in HttpSource.cs.

![image](https://user-images.githubusercontent.com/8766776/86267447-9564bc80-bb7b-11ea-967e-20a44aa299e9.png)